### ISSUE TITLE
Add support for screenrecord

### ISSUE THREAD

**Issue by qmfrederik** [quamotion/madb#55]
One of the outcomes of #40 was that we may want to add support for `screenrecord` as a high-performance way of capturing the screen of an Android device.

**Comment by Viper2k4** [comment]
That would be insane. 

I only test it a bit by my self (simply use `SendAdbRequest` to get the data and out put it to the Console). But i never worked before with VideoStreams. So i'm at point i dont how how to capture a "Frame" out of the stream.

The hole part could be even more complicated if we use H264. Because i think we would need a decoder. 

I search for a projekt that simply accept a H264-Stream and do the work for us^^. But i dont finde one :'(

**Comment by qmfrederik** [comment]
Yeah, H.264 is not something we do every day as.NET developers :)

You can have a look at (amongst others):
* FFmpeg
* Accord
* streamcoders.com

They tend to come with specific licenses so you may want to watch out for that.

You could, at first, try these things separately:
* Can you invoke screenrecord via `SendAdbRequest` and save the output to a file
* Is the file valid (e.g. can you open it with something like VLC)
* Can you use any of the H.264 decoders to extract the individual frames

And if that works, you could try to do it real time (although I’m not sure what the CPU overhead of that would be, though).

Let me know if you make any progress on this. I may have some time to look into this, but that will be a couple of sprints from now.

**Comment by Viper2k4** [comment]
Cant get a valid video.

Saving to file is no problem (the file it self is not playable with VLC or so).  FFmpeg can demuxe it but with many fragments. I try differnt options but every time the same result.

I god many error while decoding:

"P slice to large", "out if range intra chroma", "negative number of zero" and so on.

The only valid file i got is if the screen is off and there are no frames^^

**Comment by Shazter** [comment]
I found the reason why the file is corrupted, if you send the screenrecorder output to stdout. Here you have the same issue only for screencap https://stackoverflow.com/questions/13578416/read-binary-stdout-data-from-adb-shell

I tested this with my windows10 pc and android phone.

adb shell screenrecord --output-format=h264 - > E:\ffmpeg\bin\testfile.h264

Afterwards i opened my testfile.h264 with a hexeditor and replaced the values 0d 0a with 0a.

**Comment by demo7up** [comment]
what about this -> https://github.com/secile/OpenH264Lib.NET
